packages/typespec-ts/test/modularIntegration/clientNaming.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { assert } from "chai";
+import { NamingClient } from "./generated/client/naming/src/index.js";
+describe("NameAndEncodedName Client", () => {
+  let client: NamingClient;
+
+  beforeEach(() => {
+    client = new NamingClient({
+      allowInsecureConnection: true,
+      retryOptions: {
+        maxRetries: 0
+      }
+    });
+  });
+
+  it("should work with property client", async () => {
+    try {
+      const result = await client.client({ clientName: true });
+      assert.isUndefined(result);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should work with property language", async () => {
+    try {
+      const result = await client.language({ tSName: true });
+      assert.isUndefined(result);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should work with property compatibleWithEncodedName", async () => {
+    try {
+      const result = await client.compatibleWithEncodedName({
+        clientName: true
+      });
+      assert.isUndefined(result);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should work with operation", async () => {
+    try {
+      const result = await client.clientName();
+      assert.isUndefined(result);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should work with parameter", async () => {
+    try {
+      const result = await client.parameter("true");
+      assert.isUndefined(result);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should work with header request", async () => {
+    try {
+      const result = await client.request("true");
+      assert.isUndefined(result);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should work with header response", async () => {
+    try {
+      const result = await client.response({
+        onResponse: function (res) {
```

Comment:
@joheredi I remember you mentioned we should avoid `onResponse` option in modular? Am I correct?